ui/pages/setup_secure.tsx

```diff
@@ -80,12 +80,11 @@ const SetupSecure: NextPage = () => {
           },
         })
         .then(({ data }) => {
-          if ("redirect_to" in data) {
-            window.location.href = data.redirect_to as string;
-            return;
-          }
           if (data?.return_to) {
-            window.location.href = data?.return_to;
+            const flowParam = data.return_to.endsWith("/setup_complete")
+              ? ""
+              : `?flow=${data.id}`;
+            window.location.href = `./setup_complete${flowParam}`;
             return;
           }
           window.location.href = "./error";
```

Comment:
I would remove this condition, let the other page handle if there is no return_to. If you want to be safe, we can check the state of the flow (but I'm not sure if that's needed)

IMO Redirecting to the error page does not make sense